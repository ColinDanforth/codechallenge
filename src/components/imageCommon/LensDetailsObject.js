import React from 'react'

const textStyle={
  marginLeft: '30px',
  marginRight: '30px',
  marginBottom: '15px',
}

const innerTextStyle={
  marginLeft: '60px',
  marginRight: '30px',
  marginBottom: '15px',
}

const splitFeatures = (features) => {
  if(features !== null) {
    const lines = features.split('<br/>')
    return lines
  }
  return []
}

const LensDetailsObject = (props) => {
  return(
    <div>
      <p style={textStyle}>
        Lens Info:
      </p>
      <p style={innerTextStyle}>Name: {props.lensInfo.friendly_name}</p>
      {splitFeatures(props.lensInfo.features).map((feature, i) => {
        if (i === 0) {
          return <p key={i} style={innerTextStyle}>Feature: {feature}</p>
        } else {
          return <p key={i} style={innerTextStyle}>{feature}</p>
        }
      })}
    </div>
  )
}

export default LensDetailsObject
