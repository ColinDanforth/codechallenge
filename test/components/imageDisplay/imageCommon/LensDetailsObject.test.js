import React from 'react';
import LensDetailsObject from '../../../../src/components/imageCommon/LensDetailsObject';
import renderer from 'react-test-renderer';

test('LensDetailsObject should display an element with the appropriate strings present', () => {
  const testObject={
    friendly_name: "Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO",
    features: "Micro Four Thirds System<br/>24-80mm (35mm Equivalent)<br/>Aperture Range: f/2.8 to f/22",
  }

  const component = renderer.create(
    <LensDetailsObject lensInfo={testObject}/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})
