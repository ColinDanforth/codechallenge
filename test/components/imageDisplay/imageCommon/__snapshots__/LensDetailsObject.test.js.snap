// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LensDetailsObject should display an element with the appropriate strings present 1`] = `
<div>
  <p
    style={
      Object {
        "marginBottom": "15px",
        "marginLeft": "30px",
        "marginRight": "30px",
      }
    }
  >
    Lens Info:
  </p>
  <p
    style={
      Object {
        "marginBottom": "15px",
        "marginLeft": "60px",
        "marginRight": "30px",
      }
    }
  >
    Name: 
    Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO
  </p>
  <p
    style={
      Object {
        "marginBottom": "15px",
        "marginLeft": "60px",
        "marginRight": "30px",
      }
    }
  >
    Feature: 
    Micro Four Thirds System
  </p>
  <p
    style={
      Object {
        "marginBottom": "15px",
        "marginLeft": "60px",
        "marginRight": "30px",
      }
    }
  >
    24-80mm (35mm Equivalent)
  </p>
  <p
    style={
      Object {
        "marginBottom": "15px",
        "marginLeft": "60px",
        "marginRight": "30px",
      }
    }
  >
    Aperture Range: f/2.8 to f/22
  </p>
</div>
`;
